File: 33-SearchInRotatedSortedArray/33-SearchInRotatedSortedArray.java
// Last updated: 8/1/2025, 10:42:04 PM
class Solution {
    public int search(int[] nums, int target) {
        int lo=0;
        int hi=nums.length-1;
        while(lo<=hi){
            int mid=lo+(hi-lo)/2;
            if(nums[mid]==target){
                return mid;
            }else if(nums[mid]>=nums[lo]){
                if( nums[lo]<=target &&target<nums[mid] ){
                hi=mid-1;
                }
                else{
                    lo=mid+1;
                }
            }else{
                if(target>nums[mid] && target<=nums[hi])
                {
                    lo=mid+1;
                }else{
                    hi=mid-1;
                }
            }
        }
        return -1;
    }
}

File: 33-SearchInRotatedSortedArray/33-SearchInRotatedSortedArray.py
# Last updated: 8/1/2025, 10:42:09 PM
class Solution:
    def search(self, nums: List[int], target: int) -> int:
       if target in nums:
        return nums.index(target)
       else:
        return -1 
    #  k=target
    #  l,u=0,len(nums)-1
    #  m=(l+u)//2
    #  while l<=u:
    #     if (nums[m]<k):
    #         l=m+1
    #     elif nums[m]>k:
    #         u=m-1
    #         m=(l+u)//2
    #     elif nums[m]==k:
    #         return m
    #  if l>u:
    #     return -1                
     